components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOathEventListener.java

```diff
@@ -259,23 +260,32 @@ private boolean revokeTokens(String username, UserStoreManager userStoreManager)
                 log.error(errorMsg, e);
                 return true;
             }
+
+            Set<String> scopes = new HashSet<>();
             for (AccessTokenDO accessTokenDO : accessTokenDOs) {
-                //Clear cache
+                // Clear cache
                 OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser(),
                         OAuth2Util.buildScopeString(accessTokenDO.getScope()));
                 OAuthUtil.clearOAuthCache(accessTokenDO.getConsumerKey(), accessTokenDO.getAuthzUser());
                 OAuthUtil.clearOAuthCache(accessTokenDO.getAccessToken());
+
+                // Get unique scopes list
+                scopes.add(OAuth2Util.buildScopeString(accessTokenDO.getScope()));
+            }
+
+            for (String scope : scopes) {
+
```

Comment:
shall we remove unnecessary new lines in several places?

Reply:
Fixed and updated the PR.